bin/lib/utils/utils.js

```diff
@@ -65,8 +65,146 @@ function pauseForMillis( millis ){
+
+	return translations;
+}
+
+function getLatest(pubDate, cachedPubDate) {
+	return Date.parse(cachedPubDate) >= Date.parse(pubDate);
+}
+
+function processEnv(name, config={}){
+	// config = { // non-mandated keys
+	//   default: ... defult value, (specify NULL if not required, otherwise barfs by default if not set),
+	//   errorMsg: ... specify an error msg, otherwise will construct a default one,
+	//   validateString: ... specify a regex if you want to check the text format of the value
+	//   validateInteger: ... converts to Integer. Specify true, or a fn test the integer value
+	//   validateJson: ... parses JSON. Specify true, or a fn to test the parsed JSON
+	// }
+	//
+	// defaults
+	// - value is required
+	// - value is a string
+	//
+	// e.g. Util.processEnv('AWS_REGION')
+	//   means return the env param AWS_REGION, but barf if it is not set
+	// e.g. Util.processEnv('AWS_REGION', {default: null})
+	//   means return the env param AWS_REGION, or return null if it is not set
+	// e.g. Util.processEnv('AWS_REGION', {validateString: /eu/})
+	//   means return the env param AWS_REGION, but barf if it is not set or if it does not contain the string "eu"
+	// e.g. Util.processEnv('AWS_REGION_JSON', {validateJson: true})
+	//   means return the env param AWS_REGION_JSON parsed into JSON, but barf if it is not set
+
+	if ( (typeof name) !== "string") {
+		throw new Error('Error: processEnv: name is not a String');
+	}
+
+	if ( (config === null) || (config === undefined) || (typeof config) !== "object" ) {
+		throw new Error('processEnv: config is not an object');
+	}
+
+	let value = null;
+
+	if (!config.hasOwnProperty('errorMsg')) {
+		config.errorMsg = `processEnv: ${name} not set, or not set correctly`;
+	}
+
+	if (process.env.hasOwnProperty(name)) {
+		value = process.env[name];
+	} else if (config.hasOwnProperty('default')) {
+		value = config.default;
+	} else if (config.hasOwnProperty('errorMsg')) {
+		throw new Error( config.errorMsg )
+	}
+
+	let preValidationValue = value;
+
+	if (config.hasOwnProperty('validateString')) {
+		if (! value.match(config.validateString)) {
+			throw new Error(config.errorMsg);
+		}
+	}
+
+	if (config.hasOwnProperty('validateInteger')) {
+		const configErrMsg = `The configuration for env param ${name} should be Boolean true (to ensure conversion to Integer) or a function defintion (to ensure conversion to Integer, and then validate the integer value)`;
+		// accept 'true' but not 'false', or a fn definition
+		const configType = typeof config.validateInteger;
+		value = parseInt(value); // always convert to integer
+
+		if (configType === "boolean"){ // accept 'true' but not 'false'
+			if (config.validateInteger === false) {
+				throw new Error(configErrMsg);
+			}
+		} else if (configType === 'function') {
+			if(! config.validateInteger(value)) {
+					throw new Error(config.errorMsg);
+			}
+		}	else {
+			throw new Error(configErrMsg);
+		}
+	} else if(config.hasOwnProperty('validateJson')){
```

Comment:
can we check the env var starts with `{` because an integer will parse as JSON but I think in this case we want to validate it is a JSON object.

Reply:
done